File: bench/prime.r
REBOL [
    Title: "Prime"
]

maxnum: 100000
i: 2

is_prime: func [n] [
    j: 2
    r: true
    while [(j * j) < (n + 1)] [
        if ((modulo n j) == 0) [
            r: false
            break
        ]
        j: j + 1
    ]
    r
]

while [i <= maxnum] [
   if (is_prime i) [
        print i
   ]
   i: i + 1
]

File: bench/prime.js
var maxnum = 100000
var i = 2

function is_prime(n) {
    var j = 2
    while (j * j < n + 1) {
        if (n % j == 0)
            return false
        j += 1
    }
    return true
}


while (i <= maxnum) {
    if (is_prime(i)) 
        console.log(i)
    i += 1
}
